src/v/cluster/tm_tx_hash_ranges.h

```diff
@@ -172,6 +172,83 @@ struct draining_txs
     auto serde_fields() { return std::tie(id, ranges, transactions); }
 };
 
+namespace hosted_transactions {
+template<class T>
```

Comment:
can we add a c++ concept describing requirements of type `T` ?

Reply:
yeah, I can do it. do you think it's needed? the main motivation behind generic "algorithms" here was to reuse already tested tx related code with minimal changes, I couldn't use inheritance because the classes / structs are used in different serde contexts and resolved to templated to win time before there is a nicer solution so it's likely a temporary pretty closed solution (the methods are not intended to be reuse in different contexts).